core/src/main/java/nl/nn/adapterframework/pipes/Base64Pipe.java

```diff
@@ -120,25 +120,28 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 		InputStream base64 = new Base64InputStream(binaryInputStream, directionEncode, getLineLength(), lineSeparatorArray);
 
 		Message result = new Message(base64, message.copyContext().withoutSize().withCharset(directionEncode ? StandardCharsets.US_ASCII.name() : getCharset()));
-		if (getOutputTypeEnum()==OutputTypes.STRING) {
+		if (getOutputTypeEnum() == OutputTypes.STRING) {
 			try {
-				result = new Message(result.asReader());
+				result = new Message(result.asReader(), result.copyContext());
 			} catch (IOException e) {
 				throw new PipeRunException(this,"cannot open stream", e);
 			}
 		}
+		// As we wrap the input-stream, we should make sure it's not closed when the session is closed as that might close this stream before reading it.
+		message.unscheduleFromCloseOnExitOf(session);
```

Comment:
Wordt het result wel ge-scheduled? misschien expliciet doen, of even commentaartje er bij zetten als het ergens anders gebeurd :)

Reply:
> Wordt het result wel ge-scheduled? misschien expliciet doen, of even commentaartje er bij zetten als het ergens anders gebeurd :)

Goed punt. Zal ik doen.
Alleen de tests zullen wat moeten worden aangepast zodat het resultaat nog kan worden gelezen. 😅